sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala

```diff
@@ -971,6 +971,10 @@ object QueryExecutionErrors {
     new DateTimeException(s"Cannot cast $s to $to.")
   }
 
+  def cannotCastDoubleToTimestampError(d: Double, to: DataType): Throwable = {
+    new DateTimeException(s"Cannot cast $d to $to.")
```

Comment:
Instead of adding a new error here, how about merge it with `cannotCastUTF8StringToDataTypeError` and call it cannotCastToDataTypeError?